vulture/core.py

```diff
@@ -384,7 +389,11 @@ def visit_ClassDef(self, node):
 
     def visit_FunctionDef(self, node):
         for decorator in node.decorator_list:
-            if getattr(decorator, 'id', None) == 'property':
+            decorator = decorator.func if isinstance(
+                decorator, ast.Call) else decorator
+            name = decorator.attr if isinstance(
+                decorator, ast.Attribute) else decorator.id
```

Comment:
These two if statements give rise to 4 different code paths, right? Can we add tests for all 4 of them?

Reply:
It seems that the if statements always evaluate to True: https://coveralls.io/builds/18099744/source?filename=vulture/core.py (6 out of 6 test cases). Can you please add tests where the if-statement is false? Or does this never happen and we can remove the if?

Reply:
I am not quite sure how this is happening because _at least_ the `@property` case dissatisfies both the conditions.
I also tested the other test cases locally - both the False conditions are tested:

```python
In [1]: import vulture; v = vulture.Vulture()

In [2]: code = """\
   ...: @property
   ...: def rahul():
   ...:     pass
   ...: """

In [3]: v.scan(code)
decorator.id

In [4]: code = """\
   ...: @f.foobar
   ...: def rahil():
   ...:     pass
   ...: """

In [5]: v.scan(code)
Ast.Attribute

In [6]: code = """\
   ...: @f.foobar()
   ...: def rahul():
   ...:     pass
   ...: """

In [7]: v.scan(code)
Ast.call
Ast.Attribute
```

I don't know how, but it also displays lines under both `if` and `else` 6x simultaneously. 😅

Reply:
In the above experiment, I've just added `print` statements with branch names:
```python
if isinstance(decorator, ast.Call):
    decorator = decorator.func
    print("Ast.call")
if isinstance(decorator, ast.Attribute):
    name = decorator.attr
    print("Ast.Attribute")
else:
    name = decorator.id
    print("decorator.id")

```

Reply:
OK, then I guess we shouldn't care too much about the counts in coveralls.io. It would be great to know if codecov has the same problems or whether it reports the executions more accurately. Could you open an issue for this, please?

Reply:
Sure, here: https://github.com/jendrikseipp/vulture/issues/141